File: scripts/initialize-images.ts
/* eslint-disable unicorn/prefer-top-level-await */

import fs from "node:fs"
import path from "node:path"

import axios from "axios"

// 랜덤 로고 생성 함수
async function generateRandomLogo(seed: string, filePath: string) {
  const url = `https://robohash.org/${seed}.png`
  const response = await axios.get(url, { responseType: "arraybuffer" })
  fs.writeFileSync(filePath, response.data)
  console.log(`Logo saved to ${filePath}`)
}

// 랜덤 썸네일 생성 함수
async function generateRandomThumbnail(
  width: number,
  height: number,
  filePath: string,
) {
  const url = `https://picsum.photos/${width}/${height}`
  const response = await axios.get(url, { responseType: "arraybuffer" })
  fs.writeFileSync(filePath, response.data)
  console.log(`Thumbnail saved to ${filePath}`)
}

async function main() {
  const logoPath = path.join("src", "images", "icon.png")
  const thumbnailPath = path.join("src", "images", "og-thumbnail.png")

  // 로고 및 썸네일 생성
  await generateRandomLogo("bloglogo123", logoPath)
  await generateRandomThumbnail(1200, 800, thumbnailPath)
}

main().catch(error => console.error(error))

File: scripts/initialize_images.py
import secrets
from pathlib import Path

import requests

def generate_random_logo(seed: str, file_path: Path) -> None:
    """Generate a random logo and save it to the specified path."""
    url = f"https://robohash.org/{seed}.png"
    response = requests.get(url)
    response.raise_for_status()
    
    file_path.write_bytes(response.content)
    print(f"Logo saved to {file_path}")

def generate_random_thumbnail(width: int, height: int, file_path: Path) -> None:
    """Generate a random thumbnail and save it to the specified path."""
    url = f"https://picsum.photos/{width}/{height}"
    response = requests.get(url)
    response.raise_for_status()
    
    file_path.write_bytes(response.content)
    print(f"Thumbnail saved to {file_path}")

def main():
    logo_path = Path("src") / "images" / "icon.png"
    thumbnail_path = Path("src") / "images" / "og-thumbnail.png"

    # Ensure the directory exists
    logo_path.parent.mkdir(parents=True, exist_ok=True)
    thumbnail_path.parent.mkdir(parents=True, exist_ok=True)

    random_seed = secrets.token_hex(16)
    
    # Generate logo and thumbnail
    generate_random_logo(random_seed, logo_path)
    generate_random_thumbnail(1200, 800, thumbnail_path)

if __name__ == "__main__":
    main()